modules/microshift-low-latency-prepare-workload.adoc

```diff
@@ -0,0 +1,94 @@
+// Module included in the following assemblies:
+//
+// * microshift_configuring/microshift-low-latency.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-low-latency-prepare-workload_{context}"]
+= Preparing a {microshift-short} workload for low latency
+
+To take advantage of low latency, workloads running on {microshift-short} must have the `microshift-low-latency` container runtime configuration set using the `RuntimeClass` feature. The CRI-O `RuntimeClass` object is installed with the `microshift-low-latency` RPM, so only the pod annotations need to be configured.
+
+.Prerequisites
+* You installed the `microshift-low-latency` RPM package.
+* You configured workload partitioning.
+
+.Procedure
+
+. Use the following example to set the following annotations in the pod spec:
++
+[source,text]
+----
+cpu-load-balancing.crio.io: "disable"
+irq-load-balancing.crio.io: "disable"
+cpu-quota.crio.io: "disable"
+cpu-load-balancing.crio.io: "disable"
+cpu-freq-governor.crio.io: "<governor>"
+----
++
+.Example pod that runs `oslat` test:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: oslat
+  annotations:
+    cpu-load-balancing.crio.io: "disable" # <1>
+    irq-load-balancing.crio.io: "disable" # <2>
+    cpu-quota.crio.io: "disable" # <3>
+    cpu-load-balancing.crio.io: "disable" # <4>
+    cpu-freq-governor.crio.io: "<governor>" # <5>
+spec:
+  runtimeClassName: microshift-low-latency # <6>
+  containers:
+  - name: oslat
+    image: quay.io/container-perf-tools/oslat
+    imagePullPolicy: Always
+    resources:
+      requests:
+        memory: "400Mi"
+        cpu: "2"
+      limits:
+        memory: "400Mi"
+        cpu: "2"
+    env:
+    - name: tool
+      value: "oslat"
+    - name: manual
+      value: "n"
+    - name: PRIO
+      value: "1"
+    - name: delay
+      value: "0"
+    - name: RUNTIME_SECONDS
+      value: "60"
+    - name: TRACE_THRESHOLD
+      value: ""
+    - name: EXTRA_ARGS
+      value: ""
+    securityContext:
+      privileged: true
+      capabilities:
+        add:
+          - SYS_NICE
+          - IPC_LOCK
+
+----
+<1> Disables the CPU load balancing for the pod.
+<2> Opts the pod out of interrupt handling (IRQ).
+<3> Disables the CPU completely fair scheduler (CFS) quota at the pod run time.
+<4> Enable or disable C-states for each CPU. Set the value to "disable" to provide the best performance for a high-priority pod. Disabling or enabling CPU load balancing is implemented on the CRI-O level using this field and the `runtimeClassName` field value.
+<5> Sets the `cpufreq` governor for each CPU. The "performance" governor is recommended for high-priority workloads.
+<6> The `runtimeClassName` must match the name of the performance profile configured in the cluster. For example, `microshift-low-latency`. Using the `microshift-low-latency` for `runtimeClassName` and setting `cpu-load-balancing.crio.io` to "disable" disables the CPU load balancing for the pod.
```

Comment:
```suggestion
<6> The `runtimeClassName` must match the name of the performance profile configured in the cluster. For example, `microshift-low-latency`.
```

Reply:
This is per the OCP docs. Is it not true for MicroShift?

Reply:
It's true - I just felt it's out of place and redundant - we have <1> which described disabling cpu load balancing so I don't think we need to mention it in <6>.

Maybe the difference is OCP doesn't describe all annotation all at once so they need to make connection between this particular annotation and its requirement?

Reply:
OCP describes in it the docs, but theres a lot of extraneous information in that section about the NTO that doesn't apply to MicroShift. So I wanted to just make the admin aware that they have to use both of these fields to get the job done. See https://docs.openshift.com/container-platform/4.16/scalability_and_performance/low_latency_tuning/cnf-provisioning-low-latency-workloads.html#cnf-node-tuning-operator-disabling-cpu-load-balancing-for-dpdk_cnf-provisioning-low-latency

Reply:
I didn't want to link there and create confusion.

Reply:
Yes, but it makes sense in case of OCP: they're focusing on specific label in that section, whereas we list all of the possible labels and the requirement of using right runtimeClassName is common to all of them.
I just don't see how providing extra info for just one of them is helping - it might lead something thinking that they don't need runtimeClass if they're not using cpu-load-balancing label.